File: SDKs/amazonaws.com/synthetics/2017-10-11/typescript/src/sdk/models/shared/canaryscheduleoutput.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



// CanaryScheduleOutput
/** 
 * How long, in seconds, for the canary to continue making regular runs according to the schedule in the <code>Expression</code> value.
**/
export class CanaryScheduleOutput extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=DurationInSeconds" })
  durationInSeconds?: number;

  @SpeakeasyMetadata({ data: "json, name=Expression" })
  expression?: string;
}

File: SDKs/amazonaws.com/synthetics/2017-10-11/python/src/sdk/models/shared/canaryscheduleoutput.py
from dataclasses import dataclass, field
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils


@dataclass_json
@dataclass
class CanaryScheduleOutput:
    r"""CanaryScheduleOutput
    How long, in seconds, for the canary to continue making regular runs according to the schedule in the <code>Expression</code> value.
    """
    
    duration_in_seconds: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('DurationInSeconds') }})
    expression: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Expression') }})
    
